### PR TITLE
Google Tag Manager

### PR DESCRIPTION
**Pull request by kevincolten**

### CODE REVIEW

File: lms/envs/common.py [hunk 1]
```diff
@@ -830,6 +830,9 @@ def _add_microsite_dirs_to_default_template_engine(settings):
 GOOGLE_SITE_VERIFICATION_ID = None
 GOOGLE_ANALYTICS_LINKEDIN = 'GOOGLE_ANALYTICS_LINKEDIN_DUMMY'
 
+######################## GOOGLE TAG MANAGER ###########################
+GOOGLE_TAG_MANAGER_ACCOUNT = None
```
Comment (kevincolten) [inline lms/envs/common.py:834]: Do I have to set it here or is it settable in an admin panel, etc somewhere?

Comment (morenol) [reply lms/envs/common.py:834]: Yes, we can set this on the admin page

File: lms/envs/production.py [hunk 2]
```diff
@@ -741,6 +741,9 @@
 GOOGLE_ANALYTICS_LINKEDIN = AUTH_TOKENS.get('GOOGLE_ANALYTICS_LINKEDIN')
 GOOGLE_SITE_VERIFICATION_ID = ENV_TOKENS.get('GOOGLE_SITE_VERIFICATION_ID')
 
+##### GOOGLE TAG MANAGER IDS #####
+GOOGLE_TAG_MANAGER_ACCOUNT = AUTH_TOKENS.get('GOOGLE_TAG_MANAGER_ACCOUNT')
```
Comment (morenol) [inline lms/envs/production.py:745]: This value is got from the lms.auth.json file that is generated by ansible in each deployment, we can add that config to the **EDXAPP_AUTH_EXTRA** ansible variable and the lms.auth.json will be generated with that tag.

File: lms/templates/main.html [hunk 3]
```diff
@@ -158,6 +158,17 @@
     </script>
 % endif
 
+<% gtm_acct = static.get_value("GOOGLE_TAG_MANAGER_ACCOUNT", settings.GOOGLE_TAG_MANAGER_ACCOUNT) %>
```
Comment (morenol) [inline lms/templates/main.html:161]: We can also set this value in the [admin panel](https://lmsdomain/admin/site_configuration/siteconfiguration/1/change/) (site configurations), we just have to add 
...
"GOOGLE_ANALYTICS_ACCOUNT": "value",
"GOOGLE_TAG_MANAGER_ACCOUNT": "value"
}

to the json there.

**Review by morenol — APPROVED** [review]
It looks good to me, I think that we can move the changes in lms/templates/main.html to the [theme repo](https://github.com/Zollege/vandelay-openedx-themes/tree/master/edx-platform/zollege/lms/templates) but  it is OK at this way too.

Regarding the way to define the tags you can use the admin page and/or the ansible variables to define them.

**Pull request merged**